src/Abstractions/NexusMods.Abstractions.Loadouts/Loadout.cs

```diff
@@ -1,150 +1,167 @@
+            .Where(d => d.A == Revision)
+            .Select(d => d.E)
+            .StartWith(conn.Db.Find(Revision))
+            .Select(id => conn.Db.Get<Model>(id));
     }
-
-    /// <summary>
-    /// Allows you to change individual files associated with a mod in this collection.
-    /// </summary>
-    /// <param name="func">Function used to change the files of a given mod.</param>
-    /// <returns>A new loadout with the files of mods changed.</returns>
-    public Loadout AlterFiles(Func<AModFile, AModFile?> func)
+    
+    public class Model(ITransaction tx) : Entity(tx)
     {
-        return this with
+        /// <summary>
+        /// The unique identifier for this loadout, casted to a <see cref="LoadoutId"/>.
+        /// </summary>
+        public LoadoutId LoadoutId => LoadoutId.From(Id);
+        
+        /// <summary>
+        /// Gets the loadout id/txid pair for this revision of the loadout.
+        /// </summary>
+        public LoadoutRevisionId LoadoutRevisionId => new(LoadoutId, Db.BasisTxId);
+        
+        public string Name
+        {
+            get => Loadout.Name.Get(this);
+            set => Loadout.Name.Add(this, value);
+        }
+        
+        /// <summary>
+        /// Get the installation id for this loadout.
+        /// </summary>
+        public EntityId InstallationId
         {
-            Mods = Mods.Keep(m => m with
+            get => Loadout.Installation.Get(this);
+            set => Loadout.Installation.Add(this, value);
+        }
+
+        /// <summary>
+        /// The revision number for this loadout, increments by one for each change.
+        /// </summary>
+        public ulong Revision
+        {
+            get => Loadout.Revision.Get(this, 0);
+            set => Loadout.Revision.Add(this, value);
+        }
+
+        /// <summary>
+        /// Get the game installation for this loadout.
+        /// </summary>
+        public GameInstallation Installation
+        {
+            get => ServiceProvider.GetRequiredService<IGameRegistry>()
+                .Get(Loadout.Installation.Get(this));
+            set
             {
-                Files = m.Files.Keep(func)
-            })
-        };
+                var id = ServiceProvider.GetRequiredService<IGameRegistry>().GetId(value);
+                Loadout.Installation.Add(this, id);
+            }
+        }
+
+        /// <summary>
+        /// Gets all mods in this loadout.
+        /// </summary>
+        public Entities<EntityIds, Mod.Model> Mods => GetReverse<Mod.Model>(Mod.Loadout);
+        
+        /// <summary>
+        /// Gets all the files in this loadout.
+        /// </summary>
+        public Entities<EntityIds, File.Model> Files => GetReverse<File.Model>(File.Loadout);
+
+                
+        /// <summary>
+        /// Gets the mod with the given id from this loadout.
+        /// </summary>
+        public Mod.Model this[ModId idx] => Mods.First(m => m.ModId == idx);
```

Comment:
Couldn't this be optimized like so:

```csharp
var mod = Db.Get<Mod.Model>(idx.Value);
if (mod.LoadoutId != LoadoutId) throw new KeyNotFoundException();
return mod;
```

Since the new relationship is in reverse, this should work, right?

Reply:
yep, good catch.